torch/csrc/utils/nested.cpp

```diff
@@ -0,0 +1,222 @@
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/nested.h>
+
+#include <pybind11/pybind11.h>
+#include <ATen/ATen.h>
+#include <ATen/NestedTensorImpl.h>
+#include <c10/core/ScalarType.h>
+
+#include <stdexcept>
+#include <vector>
+
+namespace torch {
+namespace utils {
+
+long compute_total_size_and_sizes_tensor(
```

Comment:
Is it possible to rely on `_nested_tensor_from_tensor_list` for all the dtype, device, etc. checks?

Reply:
Looking at the implementation for both torch::tensor() and _nested_tensor_from_tensor_list, we do the same amount of checks

Reply:
Sure, but this could deduplicate code